Sources/OpenFeature/Provider/MultiProvider/FirstMatchStrategy.swift

```diff
@@ -0,0 +1,40 @@
+/// FirstMatchStrategy is a strategy that evaluates a feature flag across multiple providers
+/// and returns the first result. Skips providers that indicate they had no value due to flag not found.
+/// If any provider returns an error result other than flag not found, the error is returned.
+final public class FirstMatchStrategy: Strategy {
```

Comment:
Small nit on naming: the term "match" [is already used](https://github.com/open-feature/swift-sdk/blob/28ccd3e0f71c2a015d507b8fba1fb294770ab74e/Sources/OpenFeature/Reason.swift#L9) to indicate a matching rule has been detected for the given evaluation context / rule. Using "FirstFoundStrategy" might be a bit more consistent with the naming we use in the evaluations.

However, I understand this name has already been settled in the [kotlin sdk](https://github.com/open-feature/kotlin-sdk/pull/168/files), and we might not want to break backwards compatibility. Feel free to disregard this comment